File: app/src/main/java/com/cn29/aac/ui/masterdetail/SimpleDetailFragment.java
package com.cn29.aac.ui.masterdetail;

import android.app.Activity;
import android.os.Bundle;
import android.support.design.widget.CollapsingToolbarLayout;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.cn29.aac.R;
import com.cn29.aac.ui.base.BaseFragment;
import com.cn29.aac.ui.masterdetail.vm.SimpleMasterDetailShareViewModel;
import javax.inject.Inject;


/**
 * A fragment representing a single FeedEntry detail screen.
 * This fragment is either contained in a {@link SimpleListActivity}
 * in two-pane mode (on tablets) or a {@link SimpleDetailActivity}
 * on handsets.
 */
public class SimpleDetailFragment extends BaseFragment {

  public static final String REPO_NAME = "repo_name";
  public static final String OWNER_NAME = "owner_name";
  @Inject
  SimpleMasterDetailShareViewModel masterDetailShareViewModel;

  private View rootView;
  private String repoName;
  private String ownerName;

  /**
   * Mandatory empty constructor for the fragment manager to instantiate the
   * fragment (e.g. upon screen orientation changes).
   */
  public SimpleDetailFragment() {
  }

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    if (getArguments().containsKey(REPO_NAME) && getArguments().containsKey(OWNER_NAME)) {
      repoName = getArguments().getString(REPO_NAME);
      ownerName = getArguments().getString(OWNER_NAME);
    }
    masterDetailShareViewModel.loadRepos(ownerName, repoName).observe(this, repoResource -> {
      Activity activity = getActivity();
      CollapsingToolbarLayout appBarLayout = activity
          .findViewById(R.id.toolbar_layout);
      TextView titleView = rootView
          .findViewById(R.id.feedentry_detail);
      assert repoResource != null;
      switch (repoResource.status) {
        case SUCCESS:
          progressDialogComponent.hideLoading();
          assert repoResource.data != null;
          if (appBarLayout != null) {
            appBarLayout.setTitle(repoResource.data.name);
          }
          titleView.setText(repoResource.data.fullName);
          break;
        case ERROR:
          progressDialogComponent.hideLoading();
          break;
        case LOADING:
          progressDialogComponent.showLoading();
          break;
      }
    });
  }

  @Override
  public View onCreateView(LayoutInflater inflater, ViewGroup container,
      Bundle savedInstanceState) {
    rootView = inflater.inflate(R.layout.fragment_simple_detail, container, false);
    return rootView;
  }
}

File: app/src/main/java/com/cn29/aac/ui/masterdetail/SimpleDetailFragment.kt
package com.cn29.aac.ui.masterdetail

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.lifecycle.Observer
import com.cn29.aac.R
import com.cn29.aac.ui.base.BaseFragment
import com.cn29.aac.ui.masterdetail.vm.SimpleMasterDetailShareViewModel
import com.cn29.aac.util.Result
import com.google.android.material.appbar.CollapsingToolbarLayout
import javax.inject.Inject


class SimpleDetailFragment
    : BaseFragment() {
    @Inject
    lateinit var masterDetailShareViewModel: SimpleMasterDetailShareViewModel
    private lateinit var rootView: View
    private lateinit var repoName: String
    private lateinit var ownerName: String


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments?.containsKey(REPO_NAME) == true && arguments?.containsKey(
                        OWNER_NAME) == true) {
            repoName = arguments?.getString(REPO_NAME).toString()
            ownerName = arguments?.getString(OWNER_NAME).toString()
        }
        masterDetailShareViewModel.loadRepos(ownerName, repoName)
                .observe(this, Observer {
                    val activity: Activity = requireActivity()
                    val appBarLayout: CollapsingToolbarLayout = activity
                            .findViewById(
                                    R.id.toolbar_layout)
                    val titleView = rootView
                            .findViewById<TextView>(
                                    R.id.feedentry_detail)
                    when (it) {
                        is Result.Success -> {
                            progressDialogComponent?.hideLoading()
                            appBarLayout.title = it.data.name
                            titleView.text = it.data.fullName
                        }
                        is Result.Error -> progressDialogComponent?.hideLoading()
                        is Result.Loading -> progressDialogComponent?.showLoading()
                        else -> Unit
                    }
                })
    }

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        rootView = inflater.inflate(R.layout.fragment_simple_detail,
                                    container,
                                    false)
        return rootView
    }

    companion object {
        const val REPO_NAME = "repo_name"
        const val OWNER_NAME = "owner_name"
    }
}